include/MFixedPoint/FpS.hpp

```diff
@@ -34,398 +34,272 @@ namespace MFixedPoint {
-	bool operator >= (FpS r) const {
-		// Optimised for when numFracBits_ is the same for both
-		// operators (first if statement).
-		if(numFracBits_ == r.numFracBits_) {
-			// Q the same for both numbers
-			return rawVal_ >= r.rawVal_;
-		} else if(numFracBits_ > r.numFracBits_) {
-			// Second number has smaller Q, so result is in that precision
-			return (rawVal_ >> (numFracBits_ - r.numFracBits_)) >= r.rawVal_; 
-		} else { // numFracBits_ < r.numFracBits_		
-			// First number has smaller Q, so result is in that precision
-			return rawVal_ >= (r.rawVal_ >> (r.numFracBits_ - numFracBits_)); 
-		}
-	}
-
-	//===============================================================================================//
-	//======================================= CONVERSION METHODS ====================================//
-	//===============================================================================================//
-	
-	/// \brief		Converts the fixed-point number into an integer.
-	/// \details	Always rounds to negative infinity (66.3 becomes 66, -66.3 becomes -67).
-	/// \tparam		IntType		The return integer type.
-	template <class IntType>
-	IntType ToInt() const {
-		// Right-shift to get rid of all the decimal bits
-		// This rounds towards negative infinity
-		return (IntType)(rawVal_ >> numFracBits_);
-	}
-
-	/// \brief		Converts the fixed-point number to a float.
-	float ToFloat() const {
-		return (float)rawVal_ / (float)((BaseType)1 << numFracBits_);
-	}
-
-	/// \brief		Converts the fixed-point number to a double.
-	double ToDouble() const {
-		return (double)rawVal_ / (double)((BaseType)1 << numFracBits_);
-	}
-
-	// Explicit Conversion Operator Overloads (casts)
-	
-	/// \brief		Conversion operator from fixed-point to int32_t.
-	operator int32_t() const {
-		return ToInt<int32_t>();
-	}
-	
-	/// \brief		Conversion operator from fixed-point to int64_t.
-	operator int64_t() const {		
-		return ToInt<int64_t>();
-	}
-	
-	/// \brief		Conversion operator from fixed-point to float.
-	/// \note		Similar to double conversion.
-	operator float() const { 
-		return ToFloat();
-	}
-	
-	/// \brief		Conversion operator from fixed-point to double.
-	/// \note		Similar to float conversion.
-	operator double() const { 
-		return ToDouble();
-	}
+
+public:
+
+    //===============================================================================================//
+    //================================== CONSTRUCTORS/DESTRUCTORS ===================================//
+    //===============================================================================================//
+
+    /// \brief		Create a fixed-point value from a integer and a num. of fractional bits.
+    FpS(int32_t integer) {
+        static_assert(std::is_integral<BaseType>::value, "Integral BaseType required for FpS class.");
+        rawVal_ = integer << numFracBits_;
+    }
+
+    /// \brief		Create a fixed-point value from a double and a num. of fractional bits.
+    FpS(double dbl) {

```

Comment:
I thought you wanted to get rid of the double stuff?

Reply:
I think I kept it to make things like `1.3__fp__32` work. I guess I just didn't compile the unit tests anymore :)